src/cmd/serve.rs

```diff
@@ -81,16 +81,21 @@ fn rebuild_done_handling(broadcaster: &Sender, res: Result<()>, reload_path: &st
     }
 }
 
-fn create_new_site(interface: &str, port: &str, output_dir: &str, config_file: &str) -> Result<(Site, String)> {
+fn create_new_site(interface: &str, port: &str, output_dir: &str, base_url: Option<&str>, config_file: &str) -> Result<(Site, String)> {
     let mut site = Site::new(env::current_dir().unwrap(), config_file)?;
 
+    let base_address = match base_url {
+        Some(base_url) => format!("{}:{}", base_url, port),
+        None => format!("{}:{}", site.config.base_url, port),
+    };
     let address = format!("{}:{}", interface, port);
-    // Override the base url so links work in localhost
+
     site.config.base_url = if site.config.base_url.ends_with('/') {
-        format!("http://{}/", address)
+        format!("http://{}/", base_address)
     } else {
-        format!("http://{}", address)
+        format!("http://{}", base_address)
```

Comment:
that change is going to break `serve` for everyone. You need to keep overriding the base url if there isn't one passed to the `serve` command otherwise every link will refer to the public domain and it becomes impossible to develop locally